services/dbaas/dbaas_client.go

```diff
@@ -50,7 +50,7 @@ type Client struct {
 func NewClient(dbaasControllerAPIAddress string) *Client {
 	c := &Client{
 		l:                         logrus.WithField("component", "dbaas.Client"),
-		connM:                     new(sync.RWMutex),
+		connM:                     &sync.RWMutex{},
```

Comment:
I think mutex should be defined as value in the client (`connM sync.RWMutex`), and then we can drop this line from the constructor at all.